bpf/lib/l4.h

```diff
@@ -37,7 +37,7 @@ static __always_inline int l4_modify_port(struct __ctx_buff *ctx, int l4_off,
 					  int off, struct csum_offset *csum_off,
 					  __be16 port, __be16 old_port)
 {
-	if (csum_l4_replace(ctx, l4_off, csum_off, old_port, port, sizeof(port)) < 0)
+	if (csum_l4_replace(ctx, l4_off, csum_off, old_port, port, 0) < 0)
```

Comment:
No, it's not a bug fix. The current code works and the updated one, I've been converting it since these are the only few places left where the update-by-diff is not used. The other option would have been to bloat up l{3,4}_csum_replace() implementation for XDP with more cases. Once this overall work is complete, I'm planning to do benchmarking in general. I doubt these cases converted here make a difference overall.